gateway-js/src/schema-helper/addExtensions.ts

```diff
@@ -1,13 +1,15 @@
 import { GraphQLSchema } from 'graphql';
+
 const { version } = require('../../package.json');
 
 export function addExtensions(schema: GraphQLSchema): GraphQLSchema {
+  const apolloExtension = schema.extensions.apollo as any ?? {};
   schema.extensions = {
     ...schema.extensions,
     apollo: {
-      ...schema.extensions.apollo,
+      ...apolloExtension,
       gateway: {
-        ...schema.extensions.apollo?.gateway,
+        ...apolloExtension?.gateway,
```

Comment:
This is unsafe (`gateway` may be undefined and you can't spread undefined) but because you've set the type of `apolloExtension` to undefined, the type checker isn't helping you. I would change line 6 to be:

```
const apolloExtension = schema.extensions.apollo;
```

And then change this line to be:
```
...(apolloExtension?.gateway ?? {})
```

Reply:
Actually I have no idea why the type checker fails to catch because if I modify just line 6 I still don't see an error. Anyway, my advice on how to change it stands.

Reply:
I updated to extract all objects safely into variables